### PR TITLE
Reorganize tctl commands to not require an auth client by default

### PR DESCRIPTION
**Pull request by vapopov**
In this PR added ability to declare tctl commands which not required auth client such as version command

Related: https://github.com/gravitational/teleport/pull/47692#issuecomment-2473542741

### CODE REVIEW

**Comment by zmb3** [comment]
I don't love this approach but I'm struggling to come up with a better idea at the moment.. 🤔

**Comment by vapopov** [comment]
> I don't love this approach but I'm struggling to come up with a better idea at the moment.. 🤔

@zmb3 I tried couple other options but they also look not perfect, like to make auth client initialization as lazy loading so only commands which requires auth going to init connection, and another one was - return error if auth client failed to init, so in commands `client` might be nil
```golang
TryRun(ctx context.Context, cmd string, client *authclient.Client)
```

**Review by rosstimothy — COMMENTED** [review]
I agree with Zac, I'm not particularly fond of splitting up the commands. We could potentially expand on your suggestion to have the client be lazily constructed and instead of 
```go
TryRun(ctx context.Context, cmd string, client *authclient.Client)
```

we could do 

```go
TryRun(ctx context.Context, cmd string, clientFunc func() *authclient.Client)
```

In that scenario, the logic of building the client can stay where it is, but only be executed if the command requires an authenticated connection to the cluster.

**Comment by vapopov** [comment]
I've refactored to have separate auth client initialization after command matching
commands that do not require an authentication client: `tctl version|top|fido2|touchid|webauthnwin`

File: tool/tctl/common/acl_command.go
Comment (zmb3) [inline tool/tctl/common/acl_command.go:38]:
```suggestion
	commonclient "github.com/gravitational/teleport/tool/tctl/common/client"
```

Package names (and their aliases) should generally be all lower case. (authclient, not authClient, testing/slogtest not testing/slogTest, etc).

File: tool/tctl/common/auth_command.go
Comment (zmb3) [inline tool/tctl/common/auth_command.go:181]: There's a lot of repeated code here. Would it be simpler to run the `clientFunc` prior to the switch?

File: tool/tctl/common/client/auth.go
Comment (zmb3) [inline tool/tctl/common/client/auth.go:27]: Does tctl have a slog logger we can use so as not to write new code that depends on logrus?

File: tool/tctl/common/client/auth.go
Comment (zmb3) [inline tool/tctl/common/client/auth.go:41]:
```suggestion
	tctlcfg "github.com/gravitational/teleport/tool/tctl/common/config"
```

File: tool/tctl/common/config/profile.go
Comment (zmb3) [inline tool/tctl/common/config/profile.go:58]:
```suggestion
		return nil, trace.BadParameter("your credentials have expired, please log in using `tsh login`")
```

**Review by zmb3 — COMMENTED** [review]
I'm liking this approach better. 👍

File: tool/tctl/common/auth_command.go
Comment (vapopov) [reply tool/tctl/common/auth_command.go:181]: In this case we init auth client (start checking profile folder, establish connection, request ping) before actual matching the command has to be run. We need to identify what command it is by `FullCommand()` and then each command decides if auth client is required

File: tool/tctl/common/client/auth.go
Comment (vapopov) [reply tool/tctl/common/client/auth.go:27]: I actually moved part of the code as is, will replace to use `slog` instead

**Comment by vapopov** [comment]
would appreciate of review of this PR

**Review by Joerger — APPROVED** [review]
LGTM, approach seems reasonable. Just left a suggestion for the naming.

File: tool/tctl/common/access_request_command.go
Comment (vapopov) [reply tool/tctl/common/access_request_command.go:152]: `close` is builtin name, may be `closeFn` we have such naming in codebase, what do you think?

Comment (Joerger) [reply tool/tctl/common/access_request_command.go:152]: >close is builtin name
Huh I didn't realize that. Then `clientClose`, `closeClient`, or `closeFn` are all fine, your call.

**Comment by vapopov** [comment]
one more review by any chance

**Review by Joerger — APPROVED** [review]

File: tool/tctl/common/plugin/plugins_command.go
Comment (hugoShaka) [inline tool/tctl/common/plugin/plugins_command.go:344]: Using a mix of early return and setting command func is a bit confusing.
Can you change the Delete, Cleanup and SCIM commands to use the same `installPluginArgs` so you can have a single commandFunc invocation for everything
```
	var commandFunc func(ctx context.Context, pluginArgs installPluginArgs) error
```

I think you might have to add a few methods to their `authClient` interface (at least `GetRole` for InstallSCIM).

**Review by hugoShaka — APPROVED** [review]
LGTM once comments are resolved

**Pull request merged**
